src/it/test-projects/creedengo-java-plugin-test-project/src/main/java/org/greencodeinitiative/creedengo/java/checks/UseFilterBeforeSort.java

```diff
@@ -0,0 +1,21 @@
+package org.greencodeinitiative.creedengo.java.checks;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+class UseFilterBeforeSort {
+    UseFilterBeforeSort() {
+    }
+
+    public void manipulateStream(final List<String> list) {
+        list.stream() // Noncompliant {{Use 'filter' before 'sorted' for better efficiency.}}
+                .sorted()
+                .filter(s -> s.startsWith("A"))
+                .collect(Collectors.toList());
```

Comment:
@relivio, can you please add here the same 3rd use case present in unit test, please ?
and upgrade integration test.